src/main/java/com/codegym/jrugotom5/service/AdvertService.java

```diff
@@ -30,4 +35,21 @@ public List<AdvertDTO> getAdvertsByDateRange(LocalDate from, LocalDate to) {
                 .map(advert -> modelMapper.map(advert, AdvertDTO.class))
                 .collect(Collectors.toList());
     }
+
+    public AdvertDTO update(AdvertDTO advertDTO) {
+        Advert advert = advertRepository.findById(advertDTO.getId())
+                .orElseThrow(() -> new InvalidIdException("Advert not found with ID: " ));
+
+        Long userId = advertDTO.getUserId();
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new InvalidIdException("User not found with ID: " ));
+
+        modelMapper.getConfiguration().setSkipNullEnabled(true);
```

Comment:
This change has a global impact and will affect other methods.

Reply:
This is your subjective opinion, where are the specifics?

Reply:
What do you mean?)
Out application has only one ModelMapper bean (see AppConfig class). When our app launches Spring initializes ModelMapper with `InheritingConfiguration()` (see JavaDoc to ModelMaper).
By default this configuration has field `skipNullEnabled = Boolean.FALSE;`. And in your method you change it to "true", but never return it back to "false" again. So after invocation of your method - `skipNullEnabled` will be true till shutdown of our app. That's what i meant.

Reply:
> This is your subjective opinion, where are the specifics?

You setSkipNullEnabled here, because you dont want null values from dto to override fields in advert that store some values. So you setSkipNullEnabled(true) to avoid `if (dto.field_name != null)` for each nullable field. You should setSkipNullEnabled(false) after you done with mapping here, because model mapper is being injected into other classes and leaving setSkipNullEnabled(true) will indeed have a global impact.